### PR TITLE
astro check --watch

### PR DESCRIPTION
**Pull request by ematipico**
# Summary

RFC for https://github.com/withastro/roadmap/issues/473

# Links

- [Full Rendered proposal](https://github.com/withastro/roadmap/blob/feat/astro-check-watch/proposals/0031-astro-watch-check.md)
- [Stage 2 proposal](https://github.com/withastro/roadmap/issues/473)
- [Stage 1 proposal](https://github.com/withastro/roadmap/discussions/436)

### CODE REVIEW

File: proposals/0031-astro-watch-check.md
Comment (matthewp) [inline proposals/0031-astro-watch-check.md:92]: Talked it over with the team and I think we want to skip the experimental flag since this is a small feature. We will have it unflagged, but won't merge until it the RFC is merged.

File: proposals/0031-astro-watch-check.md
Comment (matthewp) [inline proposals/0031-astro-watch-check.md:80]: There are 3rd party tools that will rerun a command on file changes, for example the [entr](https://eradman.com/entrproject/) cli tool. Also some script runner tools, such as [wireit](https://github.com/google/wireit) have watch modes that rerun a script. 

It would be slower than this approach because all of Astro+TypeScript would reexecute each time.

File: proposals/0031-astro-watch-check.md
Comment (matthewp) [inline proposals/0031-astro-watch-check.md:62]: @ematipico we wound up not going with this strategy for testing. Mind updating the text to reflect how the testing works? I think it's ok that it's skipped for now, the RFC should reflect the _intent_ of how testing works.

File: proposals/0031-astro-watch-check.md
Comment (matthewp) [inline proposals/0031-astro-watch-check.md:69]:
```suggestion
In addition to unit tests, the feature will be manually tested with demo projects.
```

File: proposals/0031-astro-watch-check.md
Comment (matthewp) [inline proposals/0031-astro-watch-check.md:73]:
```suggestion
- This feature could be implemented in userland, but it wouldn't offer the same DX to
```

File: proposals/0031-astro-watch-check.md
Comment (matthewp) [inline proposals/0031-astro-watch-check.md:75]:
```suggestion
- The command is a long-lived process, which means the user needs to kill the command
```

File: proposals/0031-astro-watch-check.md
Comment (matthewp) [inline proposals/0031-astro-watch-check.md:92]:
```suggestion
- Create a preview release for user feedback.
- Release the feature in as a `minor` without experimental flags since it is a small improvement without many questions.
```

File: proposals/0031-astro-watch-check.md
Comment (Princesseuh) [inline proposals/0031-astro-watch-check.md:19]:
```suggestion
Add a `--watch` flag to the `astro check` command.
```

File: proposals/0031-astro-watch-check.md
Comment (Princesseuh) [inline proposals/0031-astro-watch-check.md:40]: Should be updated with the new `chokidar` method and the rationales behind going that way instead of using Vite

File: proposals/0031-astro-watch-check.md
Comment (Princesseuh) [inline proposals/0031-astro-watch-check.md:86]: This sentence is hard to read, I'm not exactly sure of the reasoning you're putting forward. Additionally, the P in LSP stands for protocol already 😛

**Review by natemoo-re — COMMENTED** [review]
Seconding the comments from Matthew and Erika, but looks good overall once those are resolved!

File: proposals/0031-astro-watch-check.md
Comment (matthewp) [inline proposals/0031-astro-watch-check.md:65]: I think this isn't true now? This paragraph can be removed since the tests aren't listening to a command.

**Review by matthewp — APPROVED** [review]

**Review by natemoo-re — APPROVED** [review]

**Review by bluwy — APPROVED** [review]

**Comment by matthewp** [comment]
Consensus was reached on this RFC and it is being merged!

**Pull request merged**
